common/src/app/common/pages/changes_builder.cljc

```diff
@@ -244,32 +244,27 @@
    (assert-page-id changes)
    (assert-objects changes)
    (let [objects (lookup-objects changes)
-
+         shapes (reverse shapes)
          set-parent-change
          (cond-> {:type :mov-objects
                   :parent-id parent-id
                   :page-id (::page-id (meta changes))
-                  :shapes (->> shapes (mapv :id))}
+                  :shapes (->> shapes reverse (mapv :id))}
```

Comment:
This implies a double reverse, don't it?

I think it's better to leave shapes as is, and reverse it in the `(reduce mk-undo-change % shapes)`